### PR TITLE
Event improvements

### PR DESCRIPTION
**Pull request by jordan-caldwell**
## Summary
This PR improves the event content type in several ways. Adds event list paragraph which allows for limiting events by type, and by upcoming or recent events. The event content type now has a page components field. A new events archive view was added.

## Metadata
| Question | Answer |
|----------|--------|
| Did you use a meaningful pull request title? | yes
| Did you apply meaningful labels to the pull request? | yes
| Did you perform a self review first? | yes
| `CHANGELOG` reflects changes? | yes
| Did you perform browser testing? | no
| Risk level | Low
| Relevant links | https://thinkoomph.jira.com/browse/RIGA-68

### CODE REVIEW

File: ecms_base/themes/custom/ecms/ecms.theme [hunk 1]
```diff
@@ -546,6 +546,68 @@ function ecms_preprocess_paragraph__numbered_step_item(&$variables): void {
...
+  if ($limit) {
+    $query->range(0, intval($limit));
+  }
+
+  // Get the current timestamp.
+  $now = time();
+
+  // Upcoming or Recent.
+  $date_condition = $paragraph->get('field_upcoming_recent')->getString();
+  if($date_condition == 'recent') {
```
Comment (ben-hamelin) [inline ecms_base/themes/custom/ecms/ecms.theme:586]: use strict comparison "===" here

File: ecms_base/themes/custom/ecms/ecms.theme [hunk 2]
```diff
@@ -546,6 +546,68 @@ function ecms_preprocess_paragraph__numbered_step_item(&$variables): void {
...
+
+  $nids = $query->execute();
+
+  // Guard against no nodes.
+  if (empty($nids)) {
+    return;
+  }
+
+  $language = \Drupal::languageManager()->getCurrentLanguage()->getId();
+  $variables['langcode'] = $language;
```
Comment (ben-hamelin) [inline ecms_base/themes/custom/ecms/ecms.theme:602]: Is this being used in the template? If not maybe we can drop it

**Review by ben-hamelin — COMMENTED** [review]
Just two minor notes in the theme file

**Comment by jordan-caldwell** [comment]
@ben-hamelin This is ready now. Had to do the export from paragraph feature package and move it over

**Comment by ben-hamelin** [comment]
Looks good!
/approved

**Pull request merged**